### DISCUSSION TITLE
Documentation suggestion

### DISCUSSION THREAD

**Discussion by eftomi** [Budibase/budibase#7580] [Ideas]
Hi, I'm exploring various possibilities of Budibase, and its similarities and differences to "traditional" development tools. The Budibase documentation is getting better and better, however there are still some catches that could be better presented. 

For example, while exploring the differences between Create and Update modes of a Form element, the documentation gives an impression that the Update mode is here just to easily populate the fields with existing values from the Repeater, while the Create mode leaves fields empty. With trial & error I found out that this Create/Update form setting also influences the Save Row action, let's say attached to a button on the same form - the first mode creates a new row, the second tries to update the row in question.

My suggestion is that in next documentation updates all the elements and their most important settings that affect the interrelations with other elements get a complete description. There's no need for lengthy explanations, it can be very condensed. 

It would be also very handy if all the exposed bindings could be listed in some table in each element's documentation page and briefly explained (e.g. "an array of ..."), to avoid a trial & error approach which takes a lot of time. 

Otherwise, kudos to everyone working on the Budibase development, it's really sleek! A skeleton of an app can be very quickly developed with it, however the trouble begins when one tries to do something more complex. Cannot wait for the Custom components :-)

**Comment by melohagan** [comment]
Hey @eftomi 

Improvements to the bindings are coming, which will include "intellisense" for available bindings as well as a descriptor of data type.

I think you may be mistaken about the form element. Create/Update is just a UI thing for the form fields. 

From this page here: https://docs.budibase.com/docs/actions#save-row

> selecting the repeater as your Data Source will make it so you can correctly get the values from the specific row.

So regardless of whether the form type is create or update, if there is a repeater involved then the save row action should always perform an update.
It would do a create row if no repeater was involved and no **_id** was provided.

**Reply by eftomi** [reply]
Thanks for the news about bindings, this is a good solution!

You are right about the form types, I tested the behavior. If I'm not mistaken, for the Create mode, if the goal is to perform an update, you can choose a repeater as a Data Source for the Save Row action. In this case the repeater provides for all the values already present in the table row, and you have to add the columns to the Save Row action for which you want to get the values directly from the form, by using the {{ _theForm_.Fields._theField_ }} binding.

The other way is that a Form itself can be a Data Source for the Save Row action. If some field is missing in the form and is not set by it, you have to add corresponding column from the repeater, like _id you mention, by the binding as in {{ _theRepeater_._theTable_._id }}

> It would do a create row if no repeater was involved and no _id was provided.

This is the crucial information and it would be nice to have it in the docs ;-)
